Kernel/TTY/TTY.cpp

```diff
@@ -55,7 +55,9 @@ void TTY::set_default_termios()
 KResultOr<size_t> TTY::read(FileDescription&, size_t, u8* buffer, size_t size)
 {
     if (Process::current()->pgid() != pgid()) {
-        Process::current()->send_signal(SIGTTIN, nullptr);
+        KResult result = Process::current()->send_signal(SIGTTIN, nullptr);
+        if (result.is_error())
+            return result;
```

Comment:
I'm not sure this is appropriate. A process should never fail to send a signal to itself (although I suppose our current slightly-dumb implementation might, since it only tries to signal the main thread of a process, instead of preferring it, and then trying any thread that doesn't block the signal...), but if for whatever reason it did, it doesn't seem right to fail `read()` with `ESRCH`.

Same comment for `TTY::write()`

Reply:
Would you like me to suppress or assert then?

Reply:
Yeah, please do :)